parser-typechecker/src/Unison/Codebase/Path.hs

```diff
@@ -104,7 +102,7 @@ parsePath' :: String -> Either String Path'
 parsePath' p = case parsePathImpl' p of
   Left e -> Left e
   Right (p, "") -> Right p
-  Right (p, rem) -> case Lexer.wordyId0 rem of
+  Right (p, rem) -> case (Lexer.wordyId0 <> Lexer.symbolyId0) rem of
```

Comment:
This allows path segments to be symbols.

Reply:
What does this semigroup do?

Reply:
This is just the `Either` semigroup. It picks the first `Right`, or the last `Left` if they're all `Left`.

Reply:
Ok, I was missing `instance Semigroup b => Semigroup (a -> b)` altogether, let alone `Semigroup (Either b)` :)